docs/source/offsets.rst

```diff
@@ -0,0 +1,50 @@
+==================
+Committing offsets
+==================
+
+Arroyo does not auto commit offsets. It is up to you to manually commit offsets when processing for that
+message is completed.
+
+The commit callback will be passed to processing strategy via `ProcessingStrategyFactory.create_with_partitions`.
+You should pass this to the strategy and have your strategy call this commit function once the rest of the message
+processing has been done.
+
+The offset to be committed in Kafka is always the next offset to be consumed from, i.e. message's offset + 1.
+In Arroyo, this means you should commit `Message.next_offset` and never `Message.offset` when done processing
+that message. Arroyo exposes `Message.position_to_commit` to make this easier.
+
+It is not safe to commit every offset in a high throughput consumer as this will add a lot of load to the system.
+Commits should generally be throttled. `CommitPolicy` is the Arroyo way of specifying commit frequency. A `CommitPolicy`
+must be passed to the stream processor, which allows specifying a minimum commit frequency (or messages between commits).
+Commit throttling can be skipped when needed (i.e. during consumer shutdown) by passing `force=True` to the commit callback.
+If you are not sure how often to commit, `ONCE_PER_SECOND` is a reasonable option.
+
+.. code-block:: Python
+
+    class ConsumerStrategy(ProcessingStrategy[KafkaPayload]):
+        def __init__(
+            self,
+            committer: Commit,
+            partitions: Mapping[Partition, int],
+        ):
+            self.__commit = commit
+
+        def poll(self) -> None:
+            pass
+
+        def submit(self, message: Message[KafkaPayload]) -> None:
+            # do something (synchronous) with the message
+            do_something()
+            self.__commit(
+                {message.partition: message.offset_to_commit}
```

Comment:
I wonder if one day we will want to change the signature of `commit` to just take the message object so there will be no more confusion on which offset to commit.

Reply:
Damn, i just realise this line is wrong

Reply:
Fwiw, I think the Position thing where the offset + timestamp gets passed every time is pretty suboptimal. It was needed because the commit log consumer is modelled as a consumer when the produce happens during the committing phase. I think this is quite wrong though and it should be modelled as a strategy instead ideally.